runtime/src/snapshot_utils.rs

```diff
@@ -1270,6 +1270,7 @@ where
     );
     let mut snapshot_archives = get_full_snapshot_archives(&snapshot_archives_dir);
     snapshot_archives.sort_unstable();
+    snapshot_archives.reverse();
```

Comment:
And the fix. Below you'll see the pop(), which is supposed to pop the oldest snapshot archive, but was popping the newest instead. Then in the for loop, with the skip, this would always skip at least 1, so the oldest snapshot archive was correctly not purged, but for the incorrect logic, hence why the tests were passing...